org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/consistencycheck/ConsistencyCheck.java

```diff
@@ -20,4 +36,110 @@ public void checkConsistency(IRepository repository) {
+		List<ConsistencyCheckErrors> errors = new ArrayList<>();
+		for (NodeTypeImplementationId id : checkForLinkedInRepository) checkRepository(errors, id, repository);
+		return errors;
+	}
+
+	private List<ConsistencyCheckErrors> checkConsistencyRelationshipTypeImplementation(Set<RelationshipTypeImplementationId> checkForLinkedInRepository, IRepository repository) {
+		// This Method gets all RelationshipTypeImplementationId from Repository.
+		List<ConsistencyCheckErrors> errors = new ArrayList<>();
+		for (RelationshipTypeImplementationId id : checkForLinkedInRepository) checkRepository(errors, id, repository);
+		return errors;
+	}
+
+	private List<ConsistencyCheckErrors> checkConsistencyArtifactType(Set<ArtifactTypeId> checkForLinkedInRepository, IRepository repository) {
+		// This Method gets all ArtifactTypeId from Repository.
+		List<ConsistencyCheckErrors> errors = new ArrayList<>();
+		for (ArtifactTypeId id : checkForLinkedInRepository) checkRepository(errors, id, repository);
+		return errors;
+	}
+
+	private List<ConsistencyCheckErrors> checkConsistencyCapabilityType(Set<CapabilityTypeId> checkForLinkedInRepository, IRepository repository) {
+		// This Method gets all CapabilityTypeId from Repository. 
+		List<ConsistencyCheckErrors> errors = new ArrayList<>();
+		for (CapabilityTypeId id : checkForLinkedInRepository) checkRepository(errors, id, repository);
+		return errors;
+	}
+
+	private List<ConsistencyCheckErrors> checkConsistencyNodeType(Set<NodeTypeId> checkForLinkedInRepository, IRepository repository) {
+		// This Method gets all NodeTypeId from Repository.
+		List<ConsistencyCheckErrors> errors = new ArrayList<>();
+		for (NodeTypeId id : checkForLinkedInRepository) checkRepository(errors, id, repository);
+		return errors;
+	}
+
+	private List<ConsistencyCheckErrors> checkConsistencyRequirementType(Set<RequirementTypeId> checkForLinkedInRepository, IRepository repository) {
+		// This Method gets all RequirementTypeId from Repository.
+		List<ConsistencyCheckErrors> errors = new ArrayList<>();
+		for (RequirementTypeId id : checkForLinkedInRepository) checkRepository(errors, id, repository);
+		return errors;
+	}
+
+	private static void checkRepository(List<ConsistencyCheckErrors> errors, DefinitionsChildId id, IRepository repository)  {
+		// This Method Iterates all linked Types for a given DefinitionsChildId.
+		Collection<DefinitionsChildId> linked;
+		try {
+			linked = repository.getReferencedDefinitionsChildIds(id);
+			// get All Linke TOSCAComponentIds
+			for (DefinitionsChildId l : linked) {
+				// Iterate all linked TOSCAComponentIds
+				if (!repository.exists(l)) { // If Repository already has the linked TOSCAComponentId in it.
+					ConsistencyCheckErrors consistencyError = new ConsistencyCheckErrors();
+					consistencyError.setCheckedId(id);
+					consistencyError.setInconsistentId(l);
+					errors.add(consistencyError); // Collect found Error.
```

Comment:
Please add statement comments always above the statement. - This concrete comment is obsolete as it just states what is written. Just remove the comment.